cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/commands/DeployClientConfigCommandRetriever.java

```diff
@@ -0,0 +1,109 @@
+import com.sequenceiq.cloudbreak.service.CloudbreakException;
+
+@Service
+public class DeployClientConfigCommandRetriever {
+
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(DeployClientConfigCommandRetriever.class);
+
+    private static final String DEPLOY_CLIENT_CONFIG_COMMAND_NAME = "DeployClusterClientConfig";
+
+    private static final int ERROR_CODES_FROM = 400;
+
+    private final ActiveCommandTableResource activeCommandTableResource;
+
+    private final RecentCommandTableResource recentCommandTableResource;
+
+    public DeployClientConfigCommandRetriever(ActiveCommandTableResource activeCommandTableResource,
+            RecentCommandTableResource recentCommandTableResource) {
+        this.activeCommandTableResource = activeCommandTableResource;
+        this.recentCommandTableResource = recentCommandTableResource;
+    }
+
+    /**
+     * Obtain deploy cluster client config command ID with different strategies:
+     * - use listActiveCommands against
+     *   CM API and find the command ID from that response.
+     * - if listActiveCommands is empty (in case of the command finishes during the
+     *   deployClusterClientConfig timeout), use /cmf/commands/activeCommandTable call, if that won't work either,
+     *   use /cmf/commands/commandTable call (both with the response cookie of listActiveCommands call)
+     */
+    public BigDecimal getCommandId(ClustersResourceApi api, Stack stack)
+            throws CloudbreakException, ApiException {
+        ApiResponse<ApiCommandList> commandListResponse =
+                api.listActiveCommandsWithHttpInfo(stack.getName(), null);
+        BigDecimal foundCommandId = getCommandIdFromActiveCommands(commandListResponse);
+        if (foundCommandId == null) {
+            LOGGER.debug("The last deploy client config command could not be found  "
+                    + "by listing active commands. Trying {} call", activeCommandTableResource.getUriPath());
+            List<CommandResource> commandsFromRunningCommandsTable = activeCommandTableResource.getCommands(
+                    api, commandListResponse.getHeaders());
+            LOGGER.debug("Processing activeCommandTable were successful.");
+            foundCommandId = getLatestDeployClientConfigCommandId(
+                    commandsFromRunningCommandsTable, activeCommandTableResource.getUriPath());
+        }
+        if (foundCommandId == null) {
+            LOGGER.debug("The last deploy client config command could not be found  "
+                    + "by listing active commandTable. Trying {} call", recentCommandTableResource.getUriPath());
+            List<CommandResource> commandsFromRecentCommandsTable = recentCommandTableResource.getCommands(
+                    api, commandListResponse.getHeaders());
+            foundCommandId = getLatestDeployClientConfigCommandId(
+                    commandsFromRecentCommandsTable, recentCommandTableResource.getUriPath());
+        }
```

Comment:
if you move out all the code from the `if` branch into a method it would be easier to see what's going on, like:
```
if(commandId.isEmpty()) {
	commandId = fetchCommandIdFromRunningCommands(...)
}
if(commandId.isEmpty()) {
	commandId = fetchCommandIdFromRecentCommands(...)
}
```